app/src/main/java/org/p2p/wallet/striga/wallet/repository/StrigaWalletRemoteRepository.kt

```diff
@@ -154,4 +161,62 @@ class StrigaWalletRemoteRepository(
             )
         }
     }
+
+    override suspend fun resendSms(
+        challengeId: StrigaWithdrawalChallengeId
+    ): StrigaDataLayerResult<Unit> {
+        return try {
+            val request = StrigaSmsResendRequest(
+                userId = strigaUserIdProvider.getUserIdOrThrow(),
+                challengeId = challengeId.value
+            )
+            api.resendSms(request)
+            return StrigaDataLayerResult.Success(Unit)
+        } catch (error: Throwable) {
+            StrigaDataLayerError.from(
+                error = error,
+                default = StrigaDataLayerError.InternalError(error)
+            )
+        }
+    }
+
+    override suspend fun verifySms(
+        smsCode: String,
+        challengeId: StrigaWithdrawalChallengeId,
+    ): StrigaDataLayerResult<Unit> {
+        return try {
+            val request = StrigaSmsVerifyRequest(
+                userId = strigaUserIdProvider.getUserIdOrThrow(),
+                challengeId = challengeId.value,
+                verificationCode = smsCode,
+                ipAddress = getLocalIpAddress()
+            )
+            api.verifySms(request)
+            return StrigaDataLayerResult.Success(Unit)
+        } catch (error: Throwable) {
+            StrigaDataLayerError.from(
+                error = error,
+                default = StrigaDataLayerError.InternalError(error)
+            )
+        }
+    }
+
+    /**
+     * This is a temporary function, until we have a proper way to get the ip address
+     */
+    private fun getLocalIpAddress(): String {
```

Comment:
hey, let's go with a separate util class like LocalIpAddressProvider so we can reuse that in case of smth

Reply:
I will move this to the StrigaIpAddressProvider, as we will obtain the actual external IP address from our striga backend